### DISCUSSION TITLE
GPIO

### DISCUSSION THREAD

**Discussion by kuolung1** [Joshua-Riek/ubuntu-rockchip#458] [General]
I use the latest version v1.29 on my orangepi 5 plus. It seems to be working very well so far, but I want to write some C++ programs to control the GPIO interface. I saw on the Internet that I need to add the wiringpi package, but use "sudo apt install The wiringpi" command shows that the package cannot be found. Please tell me how to install it, or are there any similar packages that can be used?

**Comment by Hephaistos22** [comment]
It seems like wiringpi is installed by default (at least with the latest release). I can run "sudo gpio readall" and I get the correct printout.
Note that if it's not installed for you, I believe that you'd have to type "sudo apt install wiringpi" (don't type The).

**Reply by kuolung1** [reply]
Yes, you are right. I should use the "sudo apt install wiringPi" command to install it. I tested it and it works normally. However, because in my case I need to use GPIO, a set of uart, a set of PWM and some Input / Output, after comparison, when installing the ubuntu 22.04 image provided by the OrangePi factory, you can use orangepi-config to set the usage Uart and PWM in GPIO, the current ubuntu 22.04.3 does not contain the orangepi-config package program. Originally, I wanted to move orangepi-config to your system by myself, but after carefully tracing the content of the script, I found that I need to use It seems that pwm and uart need to be modified to the kernel. In this system, how can I use PWM and UART in GPIO?
